source/extensions/geoip_providers/maxmind/geoip_provider.cc

```diff
@@ -84,8 +90,50 @@ void GeoipProviderConfig::incCounter(Stats::StatName name) {
   stats_scope_->counterFromStatName(name).inc();
 }
 
+GeoipProvider::GeoipProvider(Event::Dispatcher& dispatcher, Api::Api& api,
+                             Singleton::InstanceSharedPtr owner,
+                             GeoipProviderConfigSharedPtr config)
+    : config_(config), owner_(owner) {
+  city_db_ =
+      config_->cityDbPath() ? initMaxmindDb(config_->cityDbPath().value(), CITY_DB_TYPE) : nullptr;
+  isp_db_ =
+      config_->ispDbPath() ? initMaxmindDb(config_->ispDbPath().value(), ISP_DB_TYPE) : nullptr;
+  anon_db_ =
+      config_->anonDbPath() ? initMaxmindDb(config_->anonDbPath().value(), ANON_DB_TYPE) : nullptr;
+  mmdb_reload_dispatcher_ = api.allocateDispatcher("mmdb_reload_routine");
+  mmdb_watcher_ = dispatcher.createFilesystemWatcher();
+  mmdb_reload_thread_ = api.threadFactory().createThread(
+      [this]() -> void {
+        ENVOY_LOG_MISC(debug, "Started mmdb_reload_routine");
+        if (config_->cityDbPath()) {
+          mmdb_watcher_->addWatch(
+              config_->cityDbPath().value(), Filesystem::Watcher::Events::Modified,
```

Comment:
should `Filesystem::Watcher::Events::MovedTo` be used instead? this will align with watched directory implementation where only move/rename would trigger the watch

https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/core/v3/base.proto#envoy-v3-api-msg-config-core-v3-watcheddirectory